### PR TITLE
chore: update mui

### PR DESCRIPTION
**Pull request by aslilac**
Closes #13734 

dependabot wants to update, but isn't smart enough to group these updates

### CODE REVIEW

**Comment by Parkreiner** [comment]
I'll tick the tires a bit in about 30 minutes

File: site/src/modules/resources/AppLink/AppLink.tsx [hunk 1]
```diff
@@ -119,12 +119,11 @@ export const AppLink: FC<AppLinkProps> = ({ app, workspace, agent }) => {
         endIcon={isPrivateApp ? undefined : <ShareIcon app={app} />}
         disabled={!canClick}
         href={href}
-        target="_blank"
         css={{
           pointerEvents: canClick ? undefined : "none",
           textDecoration: "none !important",
         }}
-        onClick={async (event) => {
+        onClick={async (event: MouseEvent<HTMLElement>) => {
```
Comment (Parkreiner) [inline site/src/modules/resources/AppLink/AppLink.tsx:126]: Tried seeing what would happen if I removed the type annotation, and man, MUI really is a disaster of megamorphic abstractions, huh

**Review by buenos-nachos — APPROVED** [review]
Tried a lot of the pages and components that use MUI a lot, and didn't notice any problems. Think this is safe to merge

File: site/src/modules/resources/AppLink/AppLink.tsx (hunk 1 above)
Comment (aslilac) [reply site/src/modules/resources/AppLink/AppLink.tsx:126]: yes, yes it is :^)

**Pull request merged**